autoextract/aio/client.py

```diff
@@ -135,7 +135,14 @@ async def request():
     result.response_stats = response_stats
     if handle_retries:
         result.retry_stats = request.retry.statistics  # type: ignore
+
     agg_stats.n_results += 1
+
+    for r in result:
+        if "error" in r:
+            # https://doc.scrapinghub.com/autoextract.html#query-level
+            raise QueryError(r["query"], r["error"])
```

Comment:
Mmm, now I see the complexity. AutoExtract API supports batching (submit a list of queries in the same request). And you can receive both right results and errors in the response. But we wouldn't want to retry for all of them in the case a single one or a few fails, it would be pretty inefficient and will provide bad results in practice. 

The logic to make it work would require to create a new query only with the failed requests and retry for them. And the policy for retrials will be also much more complex, as you can have different kind of errors, and each with a different policy. 

That makes me think that it makes no sense to support `QueryErrors` in this library. Even if we do it, it won't be efficient. Imagine the user submit 100 queries in a single request, 10 returns with errors, so you start retrying them... and one of them is always a timeout, so you consume all the retrials. It can take a lot of time until you come back to the client. And the client has this request slot occupied. 

It would be much more efficient in this case if the client obtains the first response with the errors, and can pack the 10 retrials with other queries he has enqueued. 

Conclusion: query level retrials should be managed at the client. We might create some tooling in this library to simplify the work in the client.

Reply:
What about the following?
- retry single queries
- don't retry multiple queries (batches) / log a warning

Reply:
Quick update: Ivan and I have talked about this situation. Another possibility is that we could create a new standalone method for single queries that have this additional retry logic enabled.

Reply:
This thread is now outdated. Resolving.